packages/store/tests/state.spec.ts

```diff
@@ -348,20 +348,45 @@ describe('State', () => {
       expect(simplePatch(null!)(simple)).toEqual({ a: 1, b: 2 });
     });
 
+    it('should be correct when patched null to null', () => {
+      expect(simplePatch(null!)(null!)).toEqual({});
+    });
+
+    it('should avoid the whole check if you want own properties only', () => {
+      function Origin() {
+        // @ts-ignore
+        this.x = `I'm an own property`;
+      }
+
+      Origin.prototype.y = `I'm not an own property`;
+
+      // @ts-ignore
+      const patcher: typeof Origin & Origin = new Origin();
+
+      expect(patcher.x).toEqual(`I'm an own property`);
+      expect(patcher.y).toEqual(`I'm not an own property`);
+
+      const existingState: any = {};
+      const resultState: any = simplePatch(patcher)(existingState);
+
+      expect(resultState.x).toEqual(`I'm an own property`);
+      expect(resultState.y).toEqual(undefined);
+    });
+
     it('should throw exception if value is array', () => {
       try {
         const simple: string[] = ['hello'];
         simplePatch(['world'])(simple);
       } catch (e) {
-        expect(e.message).toEqual('Patching arrays is not supported.');
+        expect(e.message).toEqual(MESSAGES[CODE.PATCHING_ARRAY]());
```

Comment:
Tests should still be explicit about the message. It should not look up the expected value.
```suggestion
        expect(e.message).toEqual('Patching arrays is not supported.');
```

Reply:
![image](https://user-images.githubusercontent.com/12021443/66078682-21a16f00-e56b-11e9-8763-8dc8c2689dc0.png)

I see that we adhere to the same errors everywhere for verification

Reply:
This is an anti-pattern in tests that we should avoid.
Your expectation should always be constant and not calculated.